geoh5py/io/h5_writer.py

```diff
@@ -600,61 +598,92 @@ def write_array_attribute(
-        Add data :obj:`~geoh5py.data.data.Data.values`.
+        Add Concatenator values.
 
         :param file: Name or handle to a geoh5 file.
         :param entity: Target entity.
         :param attribute: Name of the attribute to be written to geoh5
         :param compression: Compression level for the data.
         :param values: Data values.
         """
+        if not isinstance(entity, Group):
+            raise TypeError(
+                "Entity must be a Group object; " f"got '{type(entity)}' instead."
+            )
+        # check type here
         with fetch_h5_handle(file, mode="r+") as h5file:
-            entity_handle = H5Writer.fetch_handle(h5file, entity)
-
-            if entity_handle is None:
-                return
-
-            name_map = KEY_MAP[attribute]
-            if isinstance(entity, Concatenator):
-                entity_handle = entity_handle["Concatenated Data"]
-
-                if (
-                    attribute == "concatenated_attributes"
-                    and entity.concat_attr_str == "Attributes Jsons"
-                ):
-                    name_map = entity.concat_attr_str
+            entity_handle, name_map = H5Writer.def_prepare_data_to_write(
+                h5file, entity, attribute
+            )
 
-            if name_map in entity_handle:
-                del entity_handle[name_map]
-                entity.workspace.repack = True
+            # Give the chance to fetch from file
+            values = getattr(entity, attribute, None) if values is None else values
 
-            if values is None:
-                values = getattr(
-                    entity, attribute, None
-                )  # Give the chance to fetch from file
-
-                if values is None:
-                    return
+            if values is None or entity_handle is None:
+                return
 
-                if (
-                    attribute == "concatenated_attributes"
-                    and entity.concat_attr_str == "Attributes Jsons"
-                ):
-                    values = [
-                        json.dumps(val).encode("utf-8") for val in values["Attributes"]
-                    ]
+            if (
+                attribute == "concatenated_attributes"
+                and isinstance(entity, Concatenator)
+                and entity.concat_attr_str == "Attributes Jsons"
+            ):
+                values = [
+                    json.dumps(val).encode("utf-8") for val in values["Attributes"]
+                ]
 
-            # Adding an array of values
-            if isinstance(values, dict) or isinstance(entity, CommentsData):
+            if name_map == "Attributes Jsons":
+                entity_handle.create_dataset(
+                    name_map,
+                    data=values,
+                    compression="gzip",
+                    compression_opts=compression,
+                    **kwargs,
+                )
+            elif isinstance(values, dict):
```

Comment:
```suggestion
```
Can probably remove that case if this function is dedicated to concatenated values